### PR TITLE
[SFSRAP-26] Updated Roslyn to v1.1

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: PluginGenerator.sln [hunk 1]
```diff
@@ -30,24 +25,20 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarQube.Plugins.Test.Comm
...
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ExampleAnalyzer2", "Tests\ExampleAnalyzers\ExampleAnalyzer2\ExampleAnalyzer2\ExampleAnalyzer2.csproj", "{AAAFF8DB-1438-4066-BE44-353B09CF6518}"
-EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SonarLint.Metadata", "SonarLint.Metadata\SonarLint.Metadata.csproj", "{FD1E676B-A621-494B-9E2F-083940100415}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "IntegrationTests", "Tests\IntegrationTests\IntegrationTests.csproj", "{CC8197AA-D03C-42DB-A71C-5AEEEE0D9FDD}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RoslynAnalyzer11", "Tests\ExampleAnalyzers\RoslynAnalyzer11\RoslynAnalyzer11\RoslynAnalyzer11.csproj", "{B43B3D94-F457-4BBA-A0D2-EF460F43B906}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RoslynAnalyzer10", "Tests\ExampleAnalyzers\RoslynAnalyzer10\RoslynAnalyzer10\RoslynAnalyzer10.csproj", "{AAAFF8DB-1438-4066-BE44-353B09CF6518}"
+EndProject
```
Comment (HSAR) [inline PluginGenerator.sln]: ExampleAnalyzer1 and ExampleAnalyzer2 have now been renamed to RoslynAnalyzer11 and RoslynAnalyzer10, respectively, to more accurately reflect that they now test different versions of Roslyn (before they both ran on Roslyn 1.0).

File: RoslynPluginGenerator/App.config [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="System.Reflection.Metadata" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.0.21.0" newVersion="1.0.21.0" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.CodeAnalysis.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.1.0.0" newVersion="1.1.0.0" />
+      </dependentAssembly>
+    </assemblyBinding>
```
Comment (HSAR) [inline RoslynPluginGenerator/App.config]: Binding redirects are in place to maintain backwards support for analysers built on Roslyn 1.0

File: Tests/ExampleAnalyzers/ExampleAnalyzer1/ExampleAnalyzer1.Test/ExampleAnalyzer1.Test.csproj [hunk 3]
```diff
@@ -1,133 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup>
```
Comment (HSAR) [inline Tests/ExampleAnalyzers/ExampleAnalyzer1/ExampleAnalyzer1.Test/ExampleAnalyzer1.Test.csproj]: The two projects ExampleAnalyzer1.Test and ExampleAnalyzer1.Vsix are unused and have been removed.

File: Tests/RoslynPluginGeneratorTests/App.config [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <assemblyIdentity name="System.Reflection.Metadata" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.0.21.0" newVersion="1.0.21.0" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.CodeAnalysis.Workspaces" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.1.0.0" newVersion="1.1.0.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (HSAR) [inline Tests/RoslynPluginGeneratorTests/App.config]: Three projects require the new binding redirects: SonarQube.Plugins.Roslyn.PluginGenerator, SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests, and IntegrationTests.

**Comment by HSAR** [comment]
Although there is a 1.1.1 release, 1.1.0 is what shipped with VS 2015 Update 1 and so is the version supported in this PR.

File: PluginGenerator.sln (hunk 1 above)
Comment (duncanpMS) [inline PluginGenerator.sln]: Renaming the projects makes sense. Calling them *RoslynAnalyzer1_1* and *RoslynAnalyzer1_0* would have made  the relationships to Roslyn version slightly clearer (imagine for example if we had to test against e.g. Roslyn 1.1.1 in the future), but it's not worth changing now.

**Comment by duncanpMS** [comment]
LGTM

**Pull request merged**